### PR TITLE
Fixed #29660 -- Fixed admin to 403 on POST to change view w/o permission.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/29660

### CODE REVIEW

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -3864,7 +3865,7 @@ def test_simple_inline_permissions(self):
             username='permissionuser', password='secret',
             email='vuser@example.com', is_staff=True,
         )
-        permissionuser.user_permissions.add(get_perm(Collector, get_permission_codename('view', Collector._meta)))
+        permissionuser.user_permissions.add(get_perm(Collector, get_permission_codename('change', Collector._meta)))
```
Comment (olivierdalang) [inline tests/admin_views/tests.py]: To me this change in the test doesn't look related to the change in the code ?

**Comment by jdufresne** [comment]
@olivierdalang You're right! Thank you for pointing out that use case.

Of note, when I save an inline object and I don't have _change_ permission on the parent object, I'm redirected to the admin index instead of the parent object's changelist view. Is this intentional or an oversight? As the save was successful, I would expect the user to be redirected to the same location as previous Django versions.

**Comment by jdufresne** [comment]
Per feedback above. I'm closing this.
